Lib/inspect.py

```diff
@@ -2673,13 +2682,10 @@ def _signature_from_callable(obj, *,
         # We also check that the 'obj' is not an instance of
         # types.WrapperDescriptorType or types.MethodWrapperType to avoid
         # infinite recursion (and even potential segfault)
-        call = _signature_get_user_defined_method(type(obj), '__call__')
+        call = getattr_static(type(obj), '__call__', None)
         if call is not None:
-            try:
-                sig = _get_signature_of(call)
-            except ValueError as ex:
-                msg = 'no signature found for {!r}'.format(obj)
-                raise ValueError(msg) from ex
+            call = _descriptor_get(call, obj)
+            return _get_signature_of(call)
```

Comment:
Did we get rid of this `ValueError` with the new logic?

Reply:
So when did this removed exception triggered before the change? Because at least from the search for the exception message, there are no other such error messages in this file, which means an error case that triggers this message before, now either triggers other messages, not does not trigger exception at all.